### PR TITLE
Add rule for concept exercise `annalyns-infiltration`

### LINKED ISSUE #19: Write analysis for concept exercise `annalyns-infiltration`
[Exercise source](https://github.com/exercism/elm/tree/main/exercises/concept/annalyns-infiltration).

This exercise might benefit from checking that `&&`, `||` etc are used instead of `case` statements.

### PR DESCRIPTION
**Pull request by jiegillet**
Closes #19 
[Sister PR here](https://github.com/exercism/website-copy/pull/2299).

I don't this the rules would catch all strange solutions (like someone one `&&` along with an `if` condition or something), but it should catch obvious attempts.

### CODE REVIEW

**Review by ceddlyburge — APPROVED** [review]

**Comment by jiegillet** [comment]
@exercism/maintainers-admin Can I get an approval please? 💛

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**
